clippy_lints/src/functions.rs

```diff
@@ -138,6 +155,81 @@ impl<'a, 'tcx> Functions {
         }
     }
 
+    fn check_line_number(self, cx: &LateContext, span: Span) {
+        let code_snippet = snippet(cx, span, "..");
+        let mut line_count = 0;
+        let mut in_comment = false;
+        for line in code_snippet.lines() {
+            let mut s_line = line.trim_left();
+            if in_comment {
+                if !s_line.contains("*/") { continue; }
+                in_comment = false;
+                let end_comment_loc = s_line.find("*/").expect("*/ should be present.");
+                s_line = &s_line[end_comment_loc..];
+            }
+            if s_line.is_empty() || s_line.starts_with("//") {
+                continue;
+            }
+            if s_line.contains("/*") {
+                let mut count_line: bool = !s_line.starts_with("/*");
+                let mut closing_comments = s_line.match_indices("*/");
+                let mut open_comments = s_line.match_indices("/*");
+                let mut single_comment = s_line.find("//");
+                let mut curr_idx: usize = 0;
+                loop {
```

Comment:
That's a lot of complexity. Maybe just treat lines containing multiple comment opening and closing things as non-comment lines. This is a very degenerate case that is rare in real code.

Reply:
Makes sense. So maybe just do this based on counts of the opens and closes?
multiple opens or multiple closes -> counts as a code line, out of comment.
one open, no close -> start of a multiline comment, so I'll toggle that in_comment var
one open, one close -> just check if remaining is empty or //, and increment line count if not so.